internal/icinga2/client_api.go

```diff
@@ -306,7 +318,26 @@ func (client *Client) checkMissedChanges(ctx context.Context, objType string, ca
 		var fakeEv *event.Event
 		if checkableIsMuted && attrs.Acknowledgement != AcknowledgementNone {
 			ackComment, err := client.fetchAcknowledgementComment(ctx, hostName, serviceName, attrs.AcknowledgementLastChange.Time())
-			if err != nil {
+			if errors.Is(err, fetchAcknowledgementCommentNoComment) {
+				// Unfortunately, there is no Acknowledgement object in Icinga 2, but only related runtime attributes
+				// attached to Host or Service objects. Those attributes contain no authorship. The only way to link an
+				// acknowledgement to a contact, when being fetched through the Config Objects API, is to find a
+				// matching Comment object, which contains an author field.
+				//
+				// This is not the case for the Event Stream API, where AcknowledgementSet has an author field.
+				//
+				// However, when no author is present, the Acknowledgement Event cannot be processed. Eventually, the
+				// Incident.processAcknowledgementEvent method will fail hard.
+				//
+				// As this very case only happens for catch-ups, one can at least hope that Icinga Notifications was
+				// notified about the AcknowledgementSet event through the Event Stream API when it happened. There
+				// might be a good chance that some manager was already assigned.
+
+				client.Logger.Warnw(
+					"Cannot find the comment for an acknowledgement, skipping this acknowledgement as the authorship is missing",
+					zap.String("object", objectName), zap.Error(err))
+				continue
```

Comment:
Instead of just skipping the object, please consider submitting a "basic" mute event like done there, that would at least get the mute state correct:

https://github.com/Icinga/icinga-notifications/blob/388732218bcaf20b07876c5d2fc2fe865f3857b0/internal/icinga2/client_api.go#L327-L332

Reply:
This will also allow you to keep the existing event processing flow and not skip the state change event below just because you could not find an ack comment.